app/scripts/controllers/SetUserCtrl.js

```diff
@@ -0,0 +1,16 @@
+(function() {
+    function SetUserCtrl($cookies, $location) {
+    
+    this.userName = null;
+    
+    this.setUserName = function(userName) {
+        $cookies.put('blocChatCurrentUser', this.userName)
```

Comment:
Because `this.userName` is always `null` the cookie is always set to the string "null" (because the `$cookie.put` method treats the value as a string).